integration/src/integration/java/pl/allegro/tech/hermes/integration/AsyncPublisherAutoRetryTest.java

```diff
@@ -40,10 +43,11 @@ public void before() {
     }
 
     @Test
-    public void shouldRetryPublishingMessageEnoughTimesWhenServerReturns500() {
+    public void shouldRetryPublishingMessageEnoughTimesWhenServerReturns500UsingRestTemplate() {
```

Comment:
Nit: as this test uses `AsyncRestTemplate`, maybe it should be named `shouldRetryPublishingMessageEnoughTimesWhenServerReturns500UsingAsyncRestTemplate`?